### DISCUSSION TITLE
External database installation

### DISCUSSION THREAD

**Discussion by Brand2** [zitadel/zitadel#3066] [Q&A]
Hi everyone,

I'm very interested in trying Zitadel, but there's one question I had RE: the operator approach you've employed. I've checked out `zitadelctl` and note that `takeoff` is a combination (essentially) of `operator` and `database`. Is it possible to deploy just `operator` and provide an external database (CockroachDB) for operator-spawned Zitadel instances to connect to? I have seen (looking at the `Configuration` struct [here](https://github.com/caos/zitadel/blob/eaaf76a6eb519a6583c6e70a1f5617f5a67967f6/operator/zitadel/kinds/iam/zitadel/configuration/desired.go#L10)) that it is possible to do this for the asset store.

The reason for the question is that it's common practice for where I would be deploying to run the databases on dedicated machines separate to Kubernetes, for a variety of reasons.

Thanks,

Adam

**Comment by Brand2** [comment]
I guess further to this: Is there anywhere that you have all of the environment variables which may be of interest when running Zitadel documented? I discovered [this](https://github.com/caos/zitadel/blob/56e10ecf30d6d5a18b0bcd8b67866b561c7d9fe2/cmd/zitadel/startup.yaml) file in the repo, which I can assume covers nearly every situation, but just wanted to make sure.

**Reply by eliobischof** [reply]
Hi @Brand2 

If you go with the operator, you are tied to using the operator's configuration options, see [here](https://github.com/caos/zitadel/blob/main/operator/zitadel/kinds/iam/zitadel/configuration/desired.go). Do you need any other configuration options that are not covered by the operator atm?

The other question: I'm trying to make ZITADEL run using an unmanaged database. Maybe we need a fix or two there. Coming back to you asap...

**Reply by Brand2** [reply]
I think the only option which might be missing is the ability to connect to an external CockroachDB if desired, but that would also assume that you've run the migrations against the external CockroachDB.

Another question I have is: Do you happen to have any instructions for running a local copy through Docker Compose using a built image from the container registry you guys publish to (which looks to be the GitHub container registry)? As this will be needed for my devs.

Sorry about all the questions, but I'm genuinely interested in the project and would like to know this stuff.

**Reply by Brand2** [reply]
RE: The local copy with Docker Compose, I created a separate discussion [here](https://github.com/caos/zitadel/discussions/3079) to outline where I've gotten to so far.
